src/read_bulk.hpp

```diff
@@ -43,6 +43,18 @@ size_t flatSize(const std::vector<Range>& ranges) {
     return size;
 }
 
+template <class Int, class Range>
+std::vector<Int> flattenRanges(const std::vector<Range>& ranges) {
+    std::vector<Int> result;
+    result.reserve(flatSize(ranges));
+    for (const auto& range : ranges) {
+        for (auto v = std::get<0>(range); v < std::get<1>(range); ++v) {
+            result.emplace_back(v);
+        }
+    }
+    return result;
+}
+
```

Comment:
This kinda of annoyance goes away with #319.

Reply:
let's get that merged, and then clean this one up